test/test_fake_tensor.py

```diff
@@ -2005,6 +2005,35 @@ def __torch_dispatch__(cls, func, types, args, kwargs):
         self.assertFalse(fake_wrapped_a.inner_tensor.is_cpu)
 
 
+    @unittest.skipIf(not RUN_CUDA, "requires cuda")
+    def test_offload_tensor(self):
+        from torch.nested._internal.wrappers import make_offload_tensor
+        from torch.nested._internal.offload_tensor import request_offload_all, OffloadTensor, offload_tensor_registry, OffloadTensorRegistry
+
+        with offload_tensor_registry(OffloadTensorRegistry()):
+            device_tensor = torch.ones(2, 2, 768, device="cuda")
+
+            a = make_offload_tensor(device_tensor=device_tensor)
+            # Trigger 'a' to be offloaded
+            request_offload_all()
+            make_offload_tensor(device_tensor=device_tensor.clone())
+
+            # Outer Tensor is on cuda, inner is on cpu
```

Comment:
Is the semantic that: after calling `request_offload_all`, the device_tensor is None and the host_tensor will be populated?

If so, might be good to check that before making the call, device_tensor is populated and the host_tensor is None?